### PR TITLE
Fixed force title issues

### LINKED ISSUE #95: Always show placeholder for title and add error message when empty
@amaldinesh7 _a

### PR DESCRIPTION
**Pull request by amaldinesh7**
Fixes #95 

- Added error message for the title field
- Made placeholder visible always

@labeebklatif _a please review

cc: @karthiknmenon @AbhayVAshokan

### CODE REVIEW

**Comment by AbhayVAshokan** [comment]
@amaldinesh7 the default placeholder for slash commands seems to be missing in case placeholder is not specified. Was that supposed to happen?

Screenshot 2021-12-13 at 3 52 53 PM

I also noticed this error message while scrolling.
Screenshot 2021-12-13 at 3 56 04 PM

**Comment by amaldinesh7** [comment]
@AbhayVAshokan Fixed the mentioned issues. Please have a look
@labeebklatif _a Please review.

File: lib/components/Editor/CustomExtensions/Title/ExtensionConfig.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Heading from "@tiptap/extension-heading";
+import { mergeAttributes } from "@tiptap/core";
+
+export default Heading.extend({
+  name: "title",
+  defining: true,
+
+  addAttributes() {
+    return {
+      level: {
```
Comment (labeebklatif) [inline lib/components/Editor/CustomExtensions/Title/ExtensionConfig.js:10]: Do we really need to add the attributes here explicitly? It's already given as default in the Heading extension. I think we won't need to extend this addAttributes method again unless we have any customization. Please share your thoughts

File: lib/components/Editor/index.js [hunk 2]
```diff
@@ -37,10 +38,21 @@ const Tiptap = (
 ) => {
   const [isImageUploadVisible, setImageUploadVisible] = useState(false);
 
+  let isPlaceholderActive = false;
+  if (placeholder) {
+    const placeholderKeysLength = Object.keys(placeholder).length;
+    isPlaceholderActive =
+      typeof placeholder === "string" ||
+      (placeholderKeysLength === 1
+        ? !placeholder.title
```
Comment (labeebklatif) [inline lib/components/Editor/index.js:47]: A confusion here. when `typeof placeholder === "string" ` is truthy, we are trying to access `placeholder.title` (line 47). but we should not expect a string value to have a property `title` right?

**Review by labeebklatif — COMMENTED** [review]
@amaldinesh7 Added some comments. Please share your thoughts on them

**Comment by amaldinesh7** [comment]
@labeebklatif _a Please review

**Review by labeebklatif — APPROVED** [review]
LGTM!

**Pull request merged**